upkie/cpp/actuation/RobotSimulator.cpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Inria
+
+#include "upkie/cpp/actuation/RobotSimulator.h"
+
+#include <iostream>
+
+bool RobotSimulatorClientAPI::changeDynamics(
+    int bodyUniqueId, int linkIndex,
+    struct RobotSimulatorChangeDynamicsArgs& args) {
+  b3PhysicsClientHandle sm = m_data->m_physicsClientHandle;
+  if (sm == 0) {
+    b3Warning("Not connected to physics server.");
+    return false;
+  }
+  b3SharedMemoryCommandHandle command = b3InitChangeDynamicsInfo(sm);
+  b3SharedMemoryStatusHandle statusHandle;
+
+  if (args.m_activationState >= 0) {
+    b3ChangeDynamicsInfoSetActivationState(command, bodyUniqueId,
+                                           args.m_activationState);
+  }
+  if (args.m_mass >= 0) {
+    b3ChangeDynamicsInfoSetMass(command, bodyUniqueId, linkIndex, args.m_mass);
+  }
+  if (!(args.m_localInertiaDiagonal[0] == 0 &&
```

Comment:
With floating-point values, we'd better consider inequalities (`x` is zero becomes e.g. `std::abs(x) < 1e-6` or some threshold that makes sense in context) rather than equalities.